homework/password-check/validation.cpp

```diff
@@ -1,2 +1,82 @@
+#include <algorithm>
+#include <cstdlib>
+#include <map>
+#include <string>
+#include <cctype>
+#include <regex>
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+std::map<ErrorCode, std::string> convertEnumsToString() {
+    std::map<ErrorCode, std::string> result{
+        {ErrorCode::Ok, "Ok"},
+        {ErrorCode::PasswordNeedsAtLeastNineCharacters, "Password needs to have at least nine characters"},
+        {ErrorCode::PasswordNeedsAtLeastOneNumber, "Password needs to have at least one number"},
+        {ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter, "Password needs to have at least one special character"},
+        {ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter, "Password needs to have at least one uppercase letter"},
+        {ErrorCode::PasswordsDoNotMatch, "Passwords do not match"}};
+    return result;
+}
+std::string getErrorMessage(ErrorCode er) {
+    auto res = convertEnumsToString();
+    return res.at(er);
+}
+bool doPasswordsMatch(std::string password1, std::string password2) {
+    if (password1.size() != password2.size()) {
+        return false;
+    }
+    if ((password1.empty() == password2.empty())) {
+        return true;
+    } 
+    else if (password1.size() == password2.size()) {
```

Comment:
When you use return inside `if` you don't need `if else` or `else`. Just write `if (sth) { return blabla; }`